src/precice/impl/DataContext.cpp

```diff
@@ -57,23 +59,22 @@ std::string DataContext::getMeshName() const
   return _mesh->getName();
 }
 
-int DataContext::getMeshID() const
+MeshID DataContext::getMeshID() const
 {
   PRECICE_ASSERT(_mesh);
   return _mesh->getID();
 }
 
-void DataContext::setMapping(MappingContext mappingContext, mesh::PtrData fromData, mesh::PtrData toData)
+void DataContext::addMapping(MappingContext mappingContext, mesh::PtrData fromData, mesh::PtrData toData)
 {
-  PRECICE_ASSERT(!hasMapping());
```

Comment:
This avoids by construction that a mapping in the `DataContext` is just overwritten with another mapping accidentally. I'm missing this security mechanism a bit in the new implementation. Is it, for example, allowed that the same mapping is added twice? What happens, if you do this? A corresponding unit test might also be a good idea here.

Reply:
You cannot override things in this function as `emplace_back` never overrides any content. If you add the same mapping twice, you would also execute it twice. We could safeguard against this if that's what you have in mind.

Reply:
I guess it does not make a lot of sense to add the same mapping twice, does it? If you can add a safeguard here, this would be good.